vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidc/helpers/ResponseModeParameters.kt

```diff
@@ -0,0 +1,33 @@
+package at.asitplus.wallet.lib.oidc.helpers
+
+import at.asitplus.wallet.lib.oidc.OpenIdConstants
+import kotlinx.serialization.Serializable
+
+@Serializable
+sealed interface ResponseModeParameters {
```

Comment:
Data classes cannot be inherited from

Reply:
You're absolutely right, but why not a `class` then? Then the property can be passed in the constructor, and also from the implementing classes (which may be data classes then!).
Just trying to reduce the amount of code we'll need to read that.

Reply:
> Just trying to reduce the amount of code we'll need to read that.

That's a good point!

> You're absolutely right, but why not a class then? Then the property can be passed in the constructor, and also from the implementing classes

It would be possible of course, is this the preferred way? It would create an additional backing field as mentioned in https://github.com/a-sit-plus/kmm-vc-library/pull/67/files/a2bdb213931433e6db4d792f573ce99dfeea3d44#r1610148347

> (which may be data classes then!).

They could already be data classes, just haven't thought about it :)

Reply:
DO NOT RESOLVE: Changed to data classes in commit 9b5d6a8edd2ecca2b12df110460ee847de1e893c

Reply:
What do you think about https://github.com/a-sit-plus/kmm-vc-library/commit/5efb28167f9e1b7025aec9f2b49eb2194baac899?
But why do those classes need to be `@Serializable` in the first place? I see that `AuthenticationResponsePreparationState` is serializable too, but I don't really see the need for it.

Reply:
Incorporated suggestions somewhere along the way